File: rhinocommon/snippets/py/offset-curve.py
from Rhino import *
from Rhino.DocObjects import *
from Rhino.Geometry import *
from Rhino.Input import *
from Rhino.Commands import *
from scriptcontext import doc
import rhinoscriptsyntax as rs

def RunCommand():
  rs, obj_ref = RhinoGet.GetOneObject("Select Curve", False, ObjectType.Curve)
  if rs <> Result.Success:
    return rs
  curve = obj_ref.Curve()
  if curve == None:
    return Result.Nothing

  rs, point = RhinoGet.GetPoint("Select Side", False)
  if rs <> Result.Success:
    return rs
  if point == Point3d.Unset:
    return Result.Nothing

  curves = curve.Offset(point, Vector3d.ZAxis, 1.0,
    doc.ModelAbsoluteTolerance, CurveOffsetCornerStyle.None)

  for offset_curve in curves:
    doc.Objects.AddCurve(offset_curve)

  doc.Views.Redraw()
  return Result.Success

if __name__ == "__main__":
  RunCommand()

File: rhinocommon/snippets/cs/offset-curve.cs
partial class Examples
{
  public static Result OffsetCurve(RhinoDoc doc)
  {
    ObjRef obj_ref;
    var rs = RhinoGet.GetOneObject(
      "Select Curve", false, ObjectType.Curve, out obj_ref);
    if (rs != Result.Success)
      return rs;
    var curve = obj_ref.Curve();
    if (curve == null)
      return Result.Nothing;

    Point3d point;
    rs = RhinoGet.GetPoint("Select Side", false, out point);
    if (rs != Result.Success)
      return rs;
    if (point == Point3d.Unset)
      return Result.Nothing;

    var curves = curve.Offset(point, Vector3d.ZAxis, 1.0,
      doc.ModelAbsoluteTolerance, CurveOffsetCornerStyle.None);

    foreach (var offset_curve in curves)
      doc.Objects.AddCurve(offset_curve);

    doc.Views.Redraw();
    return Result.Success;
  }
}
